packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -498,11 +498,13 @@ export class ProjectService extends BaseService {
         explores: (Explore | ExploreError)[],
     ) {
         // We delete the explores when saving to cache which cascades to the catalog
-        // So we need to get the current tagged catalog items before deleting the explores (to do a best effort re-tag)
+        // So we need to get the current tagged catalog items before deleting the explores (to do a best effort re-tag) and icons
         const prevCatalogItemsWithTags =
             await this.catalogModel.getCatalogItemsWithTags(projectUuid, {
                 onlyTagged: true, // We only need the tagged catalog items
             });
+        const prevCatalogItemsWithIcons =
+            await this.catalogModel.getCatalogItemsWithIcons(projectUuid);
```

Comment:
could we find an approach to merge these during downtime?

Reply:
Yes, I think should do it so we minimize the amount of trips to the database during the scheduled job

Reply:
cool cool, separate PR fine or do you want me to address it now?